crates/proto/src/op/message.rs

```diff
@@ -141,29 +142,23 @@ impl Message {
     /// * `op_code` - operation of the request
     /// * `response_code` - the error code for the response
     pub fn error_msg(id: u16, op_code: OpCode, response_code: ResponseCode) -> Message {
-        let mut message: Message = Message::new();
-        message.set_message_type(MessageType::Response);
-        message.set_id(id);
-        message.set_response_code(response_code);
-        message.set_op_code(op_code);
+        let mut message = Message::new();
+        message
+            .set_message_type(MessageType::Response)
+            .set_id(id)
+            .set_response_code(response_code)
+            .set_op_code(op_code);
 
         message
     }
 
     /// Truncates a Message, this blindly removes all response fields and sets truncated to `true`
     pub fn truncate(&self) -> Self {
-        let mut truncated: Message = Message::new();
-        truncated.set_id(self.id());
-        truncated.set_message_type(self.message_type());
-        truncated.set_op_code(self.op_code());
-        truncated.set_authoritative(self.authoritative());
+        let mut truncated = self.clone();
         truncated.set_truncated(true);
-        truncated.set_recursion_desired(self.recursion_desired());
-        truncated.set_recursion_available(self.recursion_available());
-        truncated.set_response_code(self.response_code());
-        if self.edns().is_some() {
-            truncated.set_edns(self.edns().unwrap().clone());
-        }
+        truncated.take_additionals();
```

Comment:
perhaps we should put a comment here just making it clear that what we're doing is dropping the various response sections from the message. As it is now, that's not an obvious side-effect.